x/execution/internal/keeper/keeper.go

```diff
@@ -469,3 +406,53 @@ func (k *Keeper) fetchEmitters(ctx sdk.Context, proc *process.Process, nodeKey s
 	}
 	return matchedRuns, nil
 }
+
+func (k *Keeper) calculatePrice(ctx sdk.Context, exec *executionpb.Execution) (sdk.Int, error) {
+	inst, err := k.instanceKeeper.Get(ctx, exec.InstanceHash)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	srv, err := k.serviceKeeper.Get(ctx, inst.ServiceHash)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	task, err := srv.GetTask(exec.TaskKey)
+	if err != nil {
+		return sdk.Int{}, err
+	}
+	perCall := sdk.NewInt(200)
+	if task.Price != nil && task.Price.PerCall != "" {
+		var ok bool
+		perCall, ok = sdk.NewIntFromString(task.Price.PerCall)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per call %q", task.Price.PerCall)
+		}
+	}
+	perSec := sdk.NewInt(20000)
+	if task.Price != nil && task.Price.PerSec != "" {
+		var ok bool
+		perSec, ok = sdk.NewIntFromString(task.Price.PerSec)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per sec %q", task.Price.PerSec)
+		}
+	}
+	perKB := sdk.NewInt(100)
+	if task.Price != nil && task.Price.PerKB != "" {
+		var ok bool
+		perKB, ok = sdk.NewIntFromString(task.Price.PerKB)
+		if !ok {
+			return sdk.Int{}, fmt.Errorf("invalid price per kb %q", task.Price.PerKB)
+		}
+	}
+	duration := sdk.NewInt(exec.GetDuration())
+	inputs, err := k.cdc.MarshalBinaryLengthPrefixed(exec.Inputs)
```

Comment:
Size of inputs/outputs based on what we actually store in db. It might not be absolutely exact as we save the whole execution so maybe there are some kind of gzip or anything that could make this data even small.
We could as well calculate the size on the whole execution but this would require that all meta data that the user doesn't control has to be paid for which is an idea that I don't really like.

Reply:
What about a simple `sizeOf`?
https://dlintw.github.io/gobyexample/public/memory-and-sizeof.html

> We could as well calculate the size on the whole execution but this would require that all meta data that the user doesn't control has to be paid for which is an idea that I don't really like.

So why forcing to encode with `MarshalBinaryLengthPrefixed` that add a lot of metadata?

Reply:
I don't think `sizeOf` is a good idea. First, it will be much easier to correlate the price to the actual data that we store so if we want to verify we can just see what's in the store. That's also the way gas is calculated, based on the message to store. Also, I'm not sure `sizeOf` behave the same way on all machines.

Reply:
as agreed, we keep the current implementation